packages/legend-query-builder/src/stores/filter/QueryBuilderFilterState.ts

```diff
@@ -195,7 +198,24 @@ export class FilterConditionState implements Hashable {
   changeOperator(val: QueryBuilderFilterOperator): void {
     this.setOperator(val);
     if (!this.operator.isCompatibleWithFilterConditionValue(this)) {
-      this.setValue(this.operator.getDefaultFilterConditionValue(this));
+      let defaultValue = this.operator.getDefaultFilterConditionValue(this);
+      if (
+        defaultValue instanceof CollectionInstanceValue &&
```

Comment:
i think we missing the case where the current value is collection and the default value is collection.
so if they switch from `is in` to `is not in`, i think we should maintain the list.

Reply:
that works already